bundles/org.openhab.binding.vektiva/src/main/java/org/openhab/binding/vektiva/internal/handler/VektivaSmarwiHandler.java

```diff
@@ -0,0 +1,263 @@
+
+    private VektivaSmarwiConfiguration config = new VektivaSmarwiConfiguration();
+
+    private final HttpClient httpClient;
+
+    private final WebSocketClient webSocketClient;
+
+    private @Nullable Session session;
+
+    private @Nullable ScheduledFuture<?> future = null;
+
+    private int lastPosition = -1;
+
+    public VektivaSmarwiHandler(Thing thing, HttpClient httpClient, WebSocketClient webSocketClient) {
+        super(thing);
+        this.httpClient = httpClient;
+        this.webSocketClient = webSocketClient;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            checkStatus();
+            return;
+        }
+
+        if (CHANNEL_CONTROL.equals(channelUID.getId())) {
+            logger.debug("Received command: {}", command);
+            String cmd = getSmarwiCommand(command);
+            if (COMMAND_OPEN.equals(cmd) || COMMAND_CLOSE.equals(cmd) || COMMAND_STOP.equals(cmd)) {
+                if (RESPONSE_OK.equals(sendCommand(cmd))) {
+                    lastPosition = COMMAND_OPEN.equals(cmd) ? 0 : 100;
+                }
+            }
+            if (command instanceof PercentType) {
+                if (RESPONSE_OK.equals(sendCommand(COMMAND_OPEN + "/" + cmd))) {
+                    lastPosition = Integer.parseInt(cmd);
+                }
+            }
+        }
+    }
+
+    @Override
+    public void dispose() {
+        super.dispose();
+        if (future != null && !(future.isCancelled() || future.isDone())) {
+            future.cancel(true);
+        }
+        closeSession();
+    }
+
+    private void closeSession() {
+        if (session != null && session.isOpen()) {
+            session.close();
+        }
+    }
+
+    private String getSmarwiCommand(Command command) {
+        switch (command.toString()) {
+            case "UP":
+                return COMMAND_OPEN;
+            case "DOWN":
+                return COMMAND_CLOSE;
+            case "STOP":
+                return COMMAND_STOP;
+            default:
+                return command.toString();
+        }
+    }
+
+    private @Nullable String sendCommand(String cmd) {
+        String url = "http://" + config.ip + "/cmd/" + cmd;
+
+        try {
+            ContentResponse resp = httpClient.newRequest(url).method(HttpMethod.GET).send();
+            logger.trace("Response: {}", resp.getContentAsString());
+            if (resp.getStatus() == 200) {
+                if (ThingStatus.ONLINE != getThing().getStatus()) {
+                    updateStatus(ThingStatus.ONLINE);
+                }
+            } else {
+                updateStatus(ThingStatus.OFFLINE);
+            }
+            return resp.getContentAsString();
+        } catch (InterruptedException e) {
+            logger.debug("API execution has been interrupted", e);
+            updateStatus(ThingStatus.OFFLINE);
```

Comment:
Is it possible to provide some additional details in the status update, that is prefered over debug logging.

Reply:
done